src/dataflow/src/render/mod.rs

```diff
@@ -146,28 +143,40 @@ mod threshold;
 mod top_k;
 mod upsert;
 
-pub(crate) fn build_local_input<A: Allocate>(
-    manager: &mut TraceManager,
-    worker: &mut TimelyWorker<A>,
-    local_inputs: &mut HashMap<GlobalId, LocalInput>,
+/// Worker-global state used during rendering.
+pub struct RenderState {
+    pub traces: TraceManager,
+    pub local_inputs: HashMap<GlobalId, LocalInput>,
+    pub ts_source_mapping: HashMap<SourceInstanceId, Weak<Option<SourceToken>>>,
+    pub ts_histories: TimestampDataUpdates,
+    pub ts_source_updates: TimestampMetadataUpdates,
+    pub dataflow_tokens: HashMap<GlobalId, Box<dyn Any>>,
+}
```

Comment:
If we have the chance to put comments to these members, that would be delightful! I only know what a few of them are.

Reply:
Heh, same! Did my best. Hopefully the commentary will continue to improve.